File: 206-reverse-linked-list/206-reverse-linked-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* reverseList(ListNode* head) {
        if(head==NULL || head->next==NULL){
            return head;
        }
        ListNode* prev=NULL,*pres=head,*nxt=head->next;
        while(pres!=NULL){
            pres->next=prev;
            prev=pres;
            pres=nxt;
            if(nxt!=NULL){
                nxt=nxt->next;
            }
        }
        head=prev;
        return head;
    }
};

File: 206-reverse-linked-list/206-reverse-linked-list.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @return {ListNode}
 */
var reverseList = function(head) {
    if(head===null || head.next===null){
        return head;
    }
    
    let prev=null;
    let present=head;
    let nxt=head.next;
    while(present!==null){
        present.next=prev;
        prev=present;
        present=nxt;
        if(nxt!==null){
            nxt=nxt.next;
        }
    }
    head=prev;
    return head;
};